racket/collects/racket/match/compiler.rkt

```diff
@@ -55,9 +59,7 @@
                                                   (Row-vars-seen row)))
                                       rows)
                                  esc)])
-      (if (procedure? predicate-stx)
-          #`[#,(predicate-stx x) rhs]
-          #`[(#,predicate-stx #,x) rhs])))
+      #'[lhs rhs]))
   (define (compile-con-pat accs pred pat-acc)
     (with-syntax* ([(tmps ...) (generate-temporaries accs)]
                    [(accs ...) accs]
```

Comment:
I considered adding syntax location to `question` here as well (GitHub won't let me comment on the right line, so you have to click expand to see it), but that seemed like it would be less useful since the location of `pred` would likely be from within this module.